service/src/main/java/com/commercetools/pspadapter/tenant/TenantFactory.java

```diff
@@ -194,13 +202,13 @@ protected NotificationDispatcher createNotificationDispatcher(TenantConfig tenan
         final NotificationProcessor defaultNotificationProcessor =
                 new DefaultNotificationProcessor(this, tenantConfig, transactionStateResolver);
 
-        final ImmutableMap.Builder<NotificationAction, NotificationProcessor> builder = ImmutableMap.builder();
-        builder.put(NotificationAction.APPOINTED, new AppointedNotificationProcessor(this, tenantConfig, transactionStateResolver));
-        builder.put(NotificationAction.CAPTURE, new CaptureNotificationProcessor(this, tenantConfig, transactionStateResolver));
-        builder.put(NotificationAction.PAID, new PaidNotificationProcessor(this, tenantConfig, transactionStateResolver));
-        builder.put(NotificationAction.UNDERPAID, new UnderpaidNotificationProcessor(this, tenantConfig, transactionStateResolver));
+        final Map<NotificationAction, NotificationProcessor> processorMap = new HashMap<>();
```

Comment:
Are we sure that in all replaced cases it is ok not to use immutable map anymore as Guavas version provided a thread safety? In case it is required the following replacement can be used:

https://www.baeldung.com/java-immutable-maps#unmodifiable-vs-immutable

Reply:
I've checked twice, immutability isn't needed as the maps are used locally or never gets modified, only to read values from.